plugins/inputs/postgresql_extensible/postgresql_extensible.go

```diff
@@ -168,22 +168,22 @@ func (p *Postgresql) Gather(acc telegraf.Accumulator) error {
 		sql_query += query_addon
 
 		if p.Query[i].Version <= db_version {
-			rows, err := db.Query(sql_query)
+			rows, err := p.DB.Query(sql_query)
```

Comment:
> maxlifetime solves the issue (of a connection being terminated by the db due to idle time) by forcing the driver to not reuse a connection after its been open for longer than the maxlifetime.

No it doesn't. For it to accomplish this goal, you would have to set the `maxlifetime` value to less than the gather interval, which means that every gather it would be getting a new connection, defeating the purpose.

> I'm not interested in doing retry logic for which this pr already has 2 builtin solutions. yes a bunch of shit can go wrong at anytime none of which was being handle prior to this PR anyways.

No, this PR is introducing new problems. I am trying to raise attention to those problems so that we don't introduce new bugs, and have people reporting issues because things aren't working right.

> Feel free to write your own PR implementing whatever crazy retry logic you want. but quite frankly if your postgresql setup needs that kind of logic you probably have bigger issues.

An idle timer and network interruption is not crazy. DBs have idle timers, network firewalls have idle timers, people have poor connections, servers have maintenance, any number of things can cause the connection to be interrupted. Retry logic is something that has to be added to any application communicating with another app.

----

In regards to the ping, I will quote specific things from those links:

https://github.com/go-sql-driver/mysql/issues/82#issuecomment-18282675
> Don't ping nor check the db connection before executing queries, it's racey.

https://github.com/go-sql-driver/mysql/issues/82#issuecomment-18283044
> The key is to understand that a DB isn't a connection. When you sql.Open()
you get a DB. It manages a pool of connections in the background, and
doesn't open any until you need them. Doing a Ping() is a fine way to do an
initial check if you want, but it's not all that Go-ish.

https://github.com/go-sql-driver/mysql/issues/82#issuecomment-18295777
> db.Ping() was introduced to give you an option to check the DSN since sql.Open(..) doesn't really open a connection: https://code.google.com/p/go/issues/detail?id=4804
Moreover it can be used to check if the server is still alive. So essentially @vadimtk is right. In many cases you "open" the DB right after you application starts and want to make sure you connection data is right. That's where db.Ping() is useful for.

https://github.com/golang/go/issues/7619#issuecomment-66092852
> what nil error from Ping() means is that "there
is at least one connection to the database which we don't yet know to be bad, or a new
connection was successfully established".  I don't see where this information would be
valuable except perhaps right after sql.Open().